### PR TITLE
EFRS-846 - created an endpoint for cloning collections and added test…

### PR DESCRIPTION
**Pull request by AleksKurilo**
…s for this case

### CODE REVIEW

**Comment by AleksKurilo** [comment]
I have read the CLA Document and I hereby sign the CLA

File: admin/src/main/java/com/exadel/frs/service/ModelService.java [hunk 1]
```diff
@@ -96,6 +100,28 @@ public Model createModel(final ModelCreateDto modelCreateDto, final String orgGu
         return modelRepository.save(model);
     }
 
+    @Transactional
+    public Model cloneModel(
```
Comment (andreigaevsky) [inline admin/src/main/java/com/exadel/frs/service/ModelService.java:104]: What about faces? We also should copy them into new model. Please, check `MigrateController.java`  and `MigrationComponent.java`

File: admin/src/main/java/com/exadel/frs/controller/ModelController.java [hunk 2]
```diff
@@ -54,14 +54,11 @@
     @ApiOperation(value = "Get model")
     public ModelResponseDto getModel(
             @ApiParam(value = "GUID of organization", required = true, example = GUID_EXAMPLE)
-            @PathVariable
-            final String orgGuid,
+            @PathVariable final String orgGuid,
```
Comment (andreigaevsky) [inline admin/src/main/java/com/exadel/frs/controller/ModelController.java:57]: Please, use `team_codestyle.xml` from the `dev` folder.

**Review by andreigaevsky — CHANGES_REQUESTED** [review]

File: admin/src/main/java/com/exadel/frs/service/ModelService.java [hunk 3]
```diff
@@ -96,6 +99,54 @@ public Model createModel(final ModelCreateDto modelCreateDto, final String orgGu
...
+
+        verifyNameIsUnique(modelCloneDto.getName(), model.getApp().getId());
+
+        val clone = new Model(model);
+        clone.setId(null);
+        clone.setName(modelCloneDto.getName());
+        clone.setAppModelAccess(new ArrayList<>());
+        val clonedModel = modelRepository.save(clone);
+
+        val feces = facesRepository.findByApiKey(model.getApiKey());
```
Comment (andreigaevsky) [inline admin/src/main/java/com/exadel/frs/service/ModelService.java:122]: Just typo. Please, rename to `faces`

File: admin/src/main/java/com/exadel/frs/service/ModelService.java [hunk 4]
```diff
@@ -96,6 +99,54 @@ public Model createModel(final ModelCreateDto modelCreateDto, final String orgGu
...
+        List<Face> cloneFaces = new ArrayList<>();
+        List<Image> cloneImages = new ArrayList<>();
+
+        for (Face face : feces) {
+            Face cloneFace = new Face(face);
```
Comment (andreigaevsky) [inline admin/src/main/java/com/exadel/frs/service/ModelService.java:127]: Please, replace with `val` everywhere

File: admin/src/main/java/com/exadel/frs/service/ModelService.java [hunk 5]
```diff
@@ -96,6 +99,54 @@ public Model createModel(final ModelCreateDto modelCreateDto, final String orgGu
...
+        val model = getModel(orgGuid, appGuid, modelGuid, userId);
+
+        authManager.verifyWritePrivilegesToApp(userId, model.getApp());
+
+        verifyNameIsUnique(modelCloneDto.getName(), model.getApp().getId());
+
+        val clone = new Model(model);
+        clone.setId(null);
+        clone.setName(modelCloneDto.getName());
+        clone.setAppModelAccess(new ArrayList<>());
```
Comment (andreigaevsky) [inline admin/src/main/java/com/exadel/frs/service/ModelService.java:119]: Maybe it's better to leave access type to a clone

**Review by andreigaevsky — CHANGES_REQUESTED** [review]

**Review by pospielov — APPROVED** [review]

**Review by andreigaevsky — APPROVED** [review]

**Pull request merged**